src/Grids/vertically_stretched_rectilinear_grid.jl

```diff
@@ -41,14 +33,86 @@ struct VerticallyStretchedRectilinearGrid{FT, TX, TY, TZ, R, A} <: AbstractRecti
-                                              size, x, y, zF,
-                                              halo = (1, 1, 1),
-                                          topology = (Periodic, Periodic, Bounded))
+"""
+    VerticallyStretchedRectilinearGrid([FT=Float64]; architecture=CPU(), size, zF,
+                                        x = nothing, y = nothing,
+                                        topology = (Periodic, Periodic, Bounded), halo = nothing)
+
+Creates a `VerticallyStretchedRectilinearGrid` with `size = (Nx, Ny, Nz)` grid points and
+a vertical grid specified by `zF`.
+
+Keyword arguments
+=================
+
+- `size` (required): A tuple prescribing the number of grid points in non-`Flat` directions.
+                     `size` is a 3-tuple for 3D models, a 2-tuple for 2D models, and either a
+                     scalar or 1-tuple for 1D models.
+
+- `topology`: A 3-tuple `(Tx, Ty, Tz)` specifying the topology of the domain.
+              `Tz` must be `Bounded` for `VerticallyStretchedRectilinearGrid`.
+              `Tx` and `Ty` specify whether the `x`- and `y`-` directions are
+              `Periodic`, `Bounded`, or `Flat`. The topology `Flat` indicates that a model does
+              not vary in that directions so that derivatives and interpolation are zero.
+              The default is `topology=(Periodic, Periodic, Bounded)`.
+
+- `architecture`: Specifies whether the array of vertical coordinates, interfaces, and spacings
+                  are stored on the CPU or GPU. Default: `architecture = CPU()`.
+
+- `zF`: An array or function of vertical index `k` that specifies the location of cell faces
+        in the `z-`direction for indices `k=1` through `k=Nz+1`, where `Nz` is the third element
+        of `size`.
+
+- `x`, `y`: Each of `x, y` are 2-tuples that specify the end points of the domain
+            in their respect directions. Scalar values may be used in `Flat` directions.
+
+- `halo`: A tuple of integers that specifies the size of the halo region of cells surrounding
+          the physical interior for each non-`Flat` direction.
+
+The physical extent of the domain can be specified via `x` and `y` keyword arguments
+indicating the left and right endpoints of each dimensions, e.g. `x=(-π, π)`.
+
+A grid topology may be specified via a tuple assigning one of `Periodic`, `Bounded, and `Flat`
+to each dimension. By default, a horizontally periodic grid topology `(Periodic, Periodic, Bounded)`
+is assumed.
+
+Constants are stored using floating point values of type `FT`. By default this is `Float64`.
+Make sure to specify the desired `FT` if not using `Float64`.
+
+Grid properties
+===============
+
+- `(Nx, Ny, Nz)::Int`: Number of physical points in the (x, y, z)-direction
+
+- `(Hx, Hy, Hz)::Int`: Number of halo points in the (x, y, z)-direction
+
+- `(Lx, Ly, Lz)::FT`: Physical extent of the grid in the (x, y, z)-direction
```

Comment:
Do we need to modify this to take out `Lz` and `z` since the vertical is dealt with separately?

Reply:
`Lz` is still a property of `VerticallyStretchedRectilinearGrid` --- I think it's inferred from `zF`?

Reply:
Yes of course. Sorry